examples/tendermint.rs

```diff
@@ -0,0 +1,230 @@
+    let sql = String::from_utf8(tx.to_vec()).unwrap();
+    return sql;
+}
+
+fn run_query_sync(sql: String) {
+    let database_url = "mysql://root:devpassword@mariadb:3306/mariadb";
+    let pool = my::Pool::new(database_url).unwrap();
+
+    info!("run_query_sync(): {}", sql);
+
+    if sql.len() > 0 {
+        let result = pool.prep_exec(sql, ());
+
+        match result {
+            Ok(_) => {
+                info!("Query successfully executed");
+            }
+            Err(_e) => {
+                info!("Query error: {}", _e);
+            }
+        }
+    }
+}
+
+struct AbciApp {
+    sql: String,
+}
+
+impl AbciApp {
+    fn new() -> AbciApp {
+        AbciApp {
+            sql: String::from(""),
+        }
+    }
+}
+
+impl abci::Application for AbciApp {
+    /// Query Connection: Called on startup from Tendermint.  The application should normally
+    /// return the last know state so Tendermint can determine if it needs to replay blocks
+    /// to the application.
+    fn info(&mut self, _req: &RequestInfo) -> ResponseInfo {
+        info!("info()");
+        ResponseInfo::new()
+    }
+
+    /// Query Connection: Set options on the application (rarely used)
+    fn set_option(&mut self, _req: &RequestSetOption) -> ResponseSetOption {
+        info!("set_option()");
+        ResponseSetOption::new()
+    }
+
+    /// Query Connection: Query your application. This usually resolves through a merkle tree holding
+    /// the state of the app.
+    fn query(&mut self, _req: &RequestQuery) -> ResponseQuery {
+        info!("query()");
+        ResponseQuery::new()
+    }
+
+    /// Consensus Connection:  Called once on startup. Usually used to establish initial (genesis)
+    /// state.
+    fn init_chain(&mut self, _req: &RequestInitChain) -> ResponseInitChain {
+        info!("init_chain()");
+        ResponseInitChain::new()
+    }
+
+    /// Consensus Connection: Called at the start of processing a block of transactions
+    /// The flow is:
+    /// begin_block()
+    ///   deliver_tx()  for each transaction in the block
+    /// end_block()
+    /// commit()
+    fn begin_block(&mut self, _req: &RequestBeginBlock) -> ResponseBeginBlock {
+        info!("begin_block()");
+        ResponseBeginBlock::new()
+    }
+
+    /// Consensus Connection: Called at the end of the block.  Often used to update the validator set.
+    fn end_block(&mut self, _req: &RequestEndBlock) -> ResponseEndBlock {
+        info!("end_block()");
+        ResponseEndBlock::new()
+    }
+
+    // Validate transactions.  Rule: SQL string must be valid SQL
+    fn check_tx(&mut self, req: &RequestCheckTx) -> ResponseCheckTx {
+        info!("check_tx()");
+
+        let sql = convert_tx(req.get_tx());
+        info!("Sql query: {}", sql);
+
+        let dialect = GenericDialect {};
+        let mut resp = ResponseCheckTx::new();
+        // TODO: cover sql injection
+        match Parser::parse_sql(&dialect, sql.clone()) {
```

Comment:
this fn should take an `&str`. smh

Reply:
lol https://docs.rs/sqlparser/0.5.0/sqlparser/parser/struct.Parser.html#method.parse_sql